### PR TITLE
lumafly: init at 3.2.0.0

### PR DESCRIPTION
**Pull request by rohanssrao**
## Description of changes

A cross platform mod manager for Hollow Knight written in Avalonia.

https://themulhima.github.io/Lumafly/

https://github.com/TheMulhima/Lumafly/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/lu/lumafly/package.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}:
+buildDotnetModule rec {
+  pname = "lumafly";
+  version = "3.2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "TheMulhima";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-oDSM5Ev9SCjbvCgDZcpzm2bVnzG04yy/WaSwJyh0b18=";
```
Comment (kirillrdy) [inline pkgs/by-name/lu/lumafly/package.nix:23]:
```suggestion
    hash = "sha256-oDSM5Ev9SCjbvCgDZcpzm2bVnzG04yy/WaSwJyh0b18=";
```

**Comment by kirillrdy** [comment]
@rohanssrao can you change commit order, otherwise first commit is referencing non existing attribute

**Comment by rohanssrao** [comment]
Done!

File: pkgs/by-name/lu/lumafly/package.nix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    icon = "lumafly";
+    comment = meta.description;
+    type = "Application";
+    categories = [ "Game" ];
+  })];
+
+  meta = with lib; {
+    description = "A cross platform mod manager for Hollow Knight written in Avalonia";
+    homepage = "https://themulhima.github.io/Lumafly/";
+    license = licenses.gpl3;
```
Comment (kirillrdy) [inline pkgs/by-name/lu/lumafly/package.nix:68]: gpl3 license is depricated

more info https://github.com/NixOS/nixpkgs/issues/304000

Comment (rohanssrao) [reply pkgs/by-name/lu/lumafly/package.nix]: Thanks, changed to `licenses.gpl3Plus` as per [the license in the repo](https://github.com/TheMulhima/Lumafly/blob/master/LICENSE#L639-L640).

File: pkgs/by-name/lu/lumafly/package.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  copyDesktopItems,
+  makeDesktopItem,
+}:
+buildDotnetModule rec {
+  pname = "lumafly";
+  version = "3.2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "TheMulhima";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/by-name/lu/lumafly/package.nix:21]:
```suggestion
    repo = "lumafly";
```

https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/lu/lumafly/package.nix [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  selfContainedBuild = true;
+
+  runtimeDeps = [
+    zlib
+    icu
+    fontconfig
+    openssl
+    xorg.libX11
+    xorg.libICE
+    xorg.libSM
```
Comment (AndersonTorres) [inline pkgs/by-name/lu/lumafly/package.nix:41]:
```suggestion
    libX11
    libICE
    libSM
```

File: pkgs/by-name/lu/lumafly/package.nix [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    desktopName = "Lumafly";
+    name = "lumafly";
+    exec = "Lumafly";
+    icon = "lumafly";
+    comment = meta.description;
+    type = "Application";
+    categories = [ "Game" ];
+  })];
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/by-name/lu/lumafly/package.nix:65]:
```suggestion
  meta = {
```

https://github.com/NixOS/nixpkgs/issues/208242

File: pkgs/by-name/lu/lumafly/package.nix [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    description = "A cross platform mod manager for Hollow Knight written in Avalonia";
```
Comment (AndersonTorres) [inline pkgs/by-name/lu/lumafly/package.nix:66]: ❤️

File: pkgs/by-name/lu/lumafly/package.nix [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    homepage = "https://themulhima.github.io/Lumafly/";
+    license = licenses.gpl3Plus;
+    mainProgram = "Lumafly";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [rohanssrao];
```
Comment (AndersonTorres) [inline pkgs/by-name/lu/lumafly/package.nix:71]:
```suggestion
    license = lib.licenses.gpl3Plus;
    mainProgram = "Lumafly";
    maintainers = with lib.maintainers; [ rohanssrao ];
    platforms = lib.platforms.linux;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/lu/lumafly/package.nix (hunk 4 above)
Comment (rohanssrao) [reply pkgs/by-name/lu/lumafly/package.nix]:
```
error: undefined variable 'libX11'
       at /home/runner/work/nixpkgs/nixpkgs/pkgs/by-name/lu/lumafly/package.nix:39:5:
           38|     openssl
           39|     libX11
             |     ^
           40|     libICE
```

File: pkgs/by-name/lu/lumafly/package.nix [hunk 8]
```diff
@@ -0,0 +1,73 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildDotnetModule,
+  dotnetCorePackages,
+  zlib,
+  icu,
+  fontconfig,
+  openssl,
+  xorg,
```
Comment (AndersonTorres) [inline pkgs/by-name/lu/lumafly/package.nix:10]:
```suggestion
  libX11,
  libICE, 
  libSM,
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by kirillrdy** [comment]
@AndersonTorres thank you for a review !

**Pull request merged**